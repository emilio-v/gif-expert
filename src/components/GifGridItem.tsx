import React from 'react';
import styled from 'styled-components';
import colors from '../utils/colors';
import { capitalize } from '../utils/capitalize/capitalize';

interface GifGridItemProps {
    id: string;
    title: string;
    url: string;
}

const Card = styled.div`
    align-items: center;
    border: 1px solid ${colors.grayishBlue};
    border-radius: 6px;
    margin-bottom: 10px;
    margin-right: 10px;
    overflow: hidden;
`;

const Image = styled.img`
    max-height: 170px;
`;
const Title = styled.p`
    padding: 5px;
    text-align: center;
`;

const GifGridItem: React.FC<GifGridItemProps> = ({ id, title, url }) => {
    return (
        <Card>
            <Image src={url} alt={title} />
            <Title>{capitalize(title)}</Title>
        </Card>
    );
};

export default GifGridItem;
