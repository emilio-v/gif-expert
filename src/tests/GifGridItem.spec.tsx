import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../components/GifGridItem';

describe('<GifGridItem>', () => {
    const mock = {
        id: 'gif-id',
        title: 'title',
        url: 'http://gif-test.com',
    };

    test('it should show GifGridItem', () => {
        const wrapper = shallow(<GifGridItem {...mock} />);
        expect(wrapper).toMatchSnapshot();
    });
});
