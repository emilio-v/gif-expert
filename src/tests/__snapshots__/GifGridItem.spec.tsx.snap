// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem> it should show GifGridItem 1`] = `
<styled.div
  className="animate__animated animate__rubberBand"
>
  <styled.img
    alt="title"
    src="http://gif-test.com"
  />
  <styled.p>
    Title
  </styled.p>
</styled.div>
`;
